src/main/java/cws/k8s/scheduler/memory/TaskScaler.java

```diff
@@ -0,0 +1,346 @@
+
+    public void afterWorkflow() {
+        if (!active) {
+            return;
+        }
+        log.debug("afterWorkflow");
+        long timestamp = System.currentTimeMillis();
+        statistics.end = timestamp;
+        log.info(statistics.summary(timestamp));
+        log.debug(statistics.exportCsv(timestamp));
+    }
+
+    /**
+     * After some testing, this was found to be the only reliable way to patch a pod
+     * using the Kubernetes client.
+     * 
+     * It will create a patch for the memory limits and request values and submit it
+     * to the cluster.
+     * 
+     * @param t          the task to be patched
+     * @param value the value to be set
+     */
+    private void patchTask(Task t, String value) {
+        String namespace = t.getPod().getMetadata().getNamespace();
+        String podname = t.getPod().getName();
+        log.debug("namespace: {}, podname: {}", namespace, podname);
+        // @formatter:off
+        String patch = "kind: Pod\n"
+                + "apiVersion: v1\n"
+                + "metadata:\n"
+                + "  name: PODNAME\n"
+                + "  namespace: NAMESPACE\n"
+                + "spec:\n"
+                + "  containers:\n"
+                + "    - name: PODNAME\n"
+                + "      resources:\n"
+                + "        limits:\n"
+                + "          memory: LIMIT\n"
+                + "        requests:\n"
+                + "          memory: REQUEST\n"
+                + "\n";
+        // @formatter:on
+        patch = patch.replace("NAMESPACE", namespace);
+        patch = patch.replace("PODNAME", podname);
+        patch = patch.replace("LIMIT", value);
+        patch = patch.replace("REQUEST", value);
+        log.debug(patch);
+
+        try {
+            client.pods().inNamespace(namespace).withName(podname).patch(patch);
+        } catch (KubernetesClientException e) {
+            // this typically happens when the feature gate InPlacePodVerticalScaling was not enabled
+            if (e.toString().contains("Forbidden: pod updates may not change fields other than")) {
+                log.error("Could not patch task. Please make sure that the feature gate 'InPlacePodVerticalScaling' is enabled in Kubernetes. See https://github.com/kubernetes/enhancements/issues/1287 for details. Task scaling will now be disabled for the rest of this workflow execution.");
+                this.active = false;
+            } else {
+                log.error("Could not patch task: {}", e);
+            }
+        }
+    }
+
+    /**
+     * This helper checks observations for sanity.
+     * 
+     * @return true is the Observation looks sane, false otherwise
+     */
+    public static boolean checkObservationSanity(Observation o) {
+        if (o.task == null || o.taskName == null || o.success == null || o.ramRequest == null || o.peakRss == null) {
+            log.error("unexpected null value in observation");
+            return false;
+        }
```

Comment:
Does the plugin activate NextflowTracing if a memory predictor is used? 
If not, please add this requirement to the docs.